src/vs/workbench/services/dialogs/common/dialogService.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import Severity from 'vs/base/common/severity';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IConfirmation, IConfirmationResult, IDialogOptions, IDialogService, IInput, IInputResult, IShowResult } from 'vs/platform/dialogs/common/dialogs';
+import { DialogsModel, IDialogsModel } from 'vs/workbench/common/dialogs';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+
+export class DialogService extends Disposable implements IDialogService {
+	_serviceBrand: undefined;
+
+	readonly model: IDialogsModel = this._register(new DialogsModel());
+
+	async confirm(confirmation: IConfirmation): Promise<IConfirmationResult> {
+		const handle = this.model.show({ confirmArgs: { confirmation } });
+		return await handle.result as IConfirmationResult;
```

Comment:
Why do we need to `as` cast here and below?

Reply:
the result is a union type but the return type can only be one